### ISSUE TITLE
Make it work for the new AP 2016 format and directory structure :|

### ISSUE THREAD

**Issue by anthonyjpesce** [datadesk/python-elections#98]

**Comment by jeremyjbowers** [comment]
Are you working on this, @anthonyjpesce? I am working on a PR for you.

**Comment by anthonyjpesce** [comment]
Not since we talked, thank you!

**Comment by palewire** [comment]
I just pushed [a patch to master](https://github.com/datadesk/python-elections/commit/769f8af4433af7bf3c26c0628126c83f44181fb7) that tries to adapt our existing framework to integrate the datestamped result files we haven't used in the past. I suspect something like this will work in a pinch.

**Comment by jeremyjbowers** [comment]
Sorry I'm so behind on this. The world's turned upside down, you know.

**Comment by palewire** [comment]
Really? Havnt heard. Off for a swim now.

**Comment by palewire** [comment]
Next we'll see if I can substitute in the ``elex`` data models for ours, which seems to be the straightest line to merging the two libraries. Wish me luck. I'm going to a dark place. https://www.youtube.com/watch?v=hLwrPDRRnf8

**Comment by jeremyjbowers** [comment]
@palewire for president. https://www.youtube.com/watch?v=6v0Q7moYmnI

**Comment by palewire** [comment]
elex Race model just slipped in without too much trouble. This only going to get uglier but it just might be possible.

**Comment by palewire** [comment]
@jeremyjbowers, if I don't survive this battle I'm counting on you to tell my story https://www.youtube.com/watch?v=DEW5p3vGo10

**Comment by eads** [comment]
If the models line up, the CLI should Just Work(tm).

On Tue, Jan 26, 2016 at 4:03 PM, Ben Welsh <notifications@github.com> wrote:

> @jeremyjbowers <https://github.com/jeremyjbowers>, if I don't survive
> this battle I'm counting on you to tell my story
> https://www.youtube.com/watch?v=DEW5p3vGo10
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datadesk/python-elections/issues/98#issuecomment-175229201>
> .
>



-- 
David Eads | http://recoveredfactory.net

"Medical statistics will be our standard of measurement: we will weigh life
for life and see where the dead lie thicker, among the workers or among the
privileged." -- Rudolf Virchow

**Comment by palewire** [comment]
That's my hope. My stopping point for today might be just refactoring them into the existing library here but I agree that theoretically it's possible after that. 

Though there are some init functions you guys are calling I don't totally understand just yet that might cause an issue.

Also, when this is done one of the most important things will be to proof how accurately I've mapped the fields from the FTP to your schema. It's the same data of course, but the CSV headers from the FTP are of course slightly different and I'm mapping a bunch at once.

**Comment by palewire** [comment]
Okay. I have a basic refactor stuffing all the data in the elex classes. Many things are probably broken, and it isn't even close to integration with your ``Election`` class, but it is possible, I think, to make an alernative call to our library and have it "serialize" out data for an application with which will try tomorrow during the AP test as a fallback source of data.

**Comment by palewire** [comment]
![today](http://a.abcnews.go.com/images/US/HT_bald_eagle_egg_hatches_jt_150325_12x5_1600.jpg)

**Comment by palewire** [comment]
Alrighty, after today's AP data test I believe we have squashed the most glaring bugs from my work yesterday. We ran two separate crons, one for the API and one for the FTP, using this simple fork in our download command.
```python
            if options['ftp']:
                logger.debug("Pulling data from FTP")
                election = Election(
                    election_date=date,
                    username='myuser',
                    password='mypass',
                )
            else:
                logger.debug("Pulling data from web API")
                election = api.Election(
                    electiondate=date,
                    testresults=True,
                    liveresults=False,
                    is_test=True
                )
```

Since python-elections is now returning the same classes as elex, the rest of our code remained 100% the same. Everything worked. Results came out identical (as far as we could tell) in our current front end application.

This is probably enough of a hack to get through Iowa and "work" as a fallback.

Taking the next step of fully integrating the FTP calls within elex is probably within reach, but I'm reluctant to dive into that refactoring without guidance from @jeremyjbowers and @eads. Do you have any opinions on how our classes here should be merged with your existing code?

**Comment by jeremyjbowers** [comment]
This is fantastic. I'll dig in on this tonight.


contact([{ "twitter": "@jeremybowers", "phone": "727-667-0510", "github": "
http://github.com/jeremyjbowers" }])

**Comment by palewire** [comment]
Okay. I am going to shave away a little bit at it today, trying to bring our big class a few baby steps closer to yours, but not matter what we do I think there's probably a difficult merge ahead.

**Comment by palewire** [comment]
![](http://www.skfny.com/wp-content/uploads/FLIGHT-OF-FREEDOM-BALD-EAGLE9.jpg)

**Comment by eads** [comment]
![bitmoji](https://da8lb468m8h1w.cloudfront.net/v2/cpanel/9174853-118011287_5-s1-v1.png?palette=1&width=246)

**Comment by jeremyjbowers** [comment]
OMFG it is totally working.

**Comment by jeremyjbowers** [comment]
```e = Election(electiondate=2016-02-01, username="****", password="****")
In [2]: len(e.candidate_reporting_units)
Out[2]: 1800```

**Comment by Schwanksta** [comment]
![Caw](http://static4.businessinsider.com/image/566ed67eeab8eab1558b4569-1190-625/eagle-handler-explains-what-happened-when-his-bald-eagle-attacked-trump.jpg)

**Comment by palewire** [comment]
So you merged it somehow?

**Comment by jeremyjbowers** [comment]
Example loader script here: https://github.com/newsdev/elex-ftp-loader

**Comment by jeremyjbowers** [comment]
@palewire Did not try to do the gigantic merge for IA/NH. I'm cool with having this be separate for now. It is a fallback, after all.

**Comment by palewire** [comment]
I agree that now is a good time to punt. Should we aim to try to land it for South Carolina?

**Comment by palewire** [comment]
![](http://vancouverislandbirds.com/110305eagle280.jpg)

**Comment by palewire** [comment]
I am going to broaden this ticket's name to aim for creating something that is closer to snapping into elex. First off, I'd like to get my tests working again so I'm not flying blind.
